### PR TITLE
Pfappserver Updates

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/components/pfFieldRuleSyslogParserRegex.vue [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  created () {
+    this.emitValidations()
+  }
+}
+</script>
+
+<style lang="scss">
+.pf-field-rule {
+  .collapse-handle {
+    cursor: pointer;
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfFieldRuleSyslogParserRegex.vue:292]: Already in `pfFieldRule.vue`. Could be removed or moved to `styles/_forms.scss`.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCaptivePortal.js [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+                collapseObject: true,
+                trackBy: 'value',
+                label: 'text',
+                options: pfConfigurationCaptivePortalChosenUnits
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('IMG path'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCaptivePortal.js]: I prefer *Image path* or *Detection image path*. I would also put it just bellow `network_detection_ip`.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCaptivePortal.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+          text: i18n.t('This is the path where the gif is on the webserver to detect if the network accesshas been enabled.'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCaptivePortal.js]: Missing space (*accesshas*).

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCiscoMobilityServicesEngine.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+} = require('vuelidate/lib/validators')
+
+export const pfConfigurationCiscoMobilityServicesEngineViewFields = (context = {}) => {
+  return [
+    {
+      tab: null,
+      fields: [
+        {
+          label: i18n.t('Enable MSE'),
+          text: i18n.t('Enable MSE.'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCiscoMobilityServicesEngine.js]: Remove the text as it's the same thing as the label.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCiscoMobilityServicesEngine.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+              component: pfFormRangeToggle,
+              attrs: {
+                values: { checked: 'enabled', unchecked: 'disabled' }
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('URL of MSE service'),
+          text: i18n.t('URL of MSE service.'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationCiscoMobilityServicesEngine.js]: Remove the text as it's the same thing as the label.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogForwarders.js [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+              validators: {
+                [i18n.t('Value required.')]: required,
+                [i18n.t('Maximum 255 characters.')]: maxLength(255),
+                [i18n.t('Syslog Forwarder exists.')]: not(and(required, conditional(isNew || isClone), hasSyslogForwarders, syslogForwarderExists))
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('Proto'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogForwarders.js]: Please expand *Proto* to *Protocol*.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogForwarders.js [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+          fields: [
+            {
+              key: 'proto',
+              component: pfFormChosen,
+              attrs: {
+                collapseObject: true,
+                placeholder: i18n.t('Click to add a proto'),
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogForwarders.js]: Please expand *proto* to *protocol*

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogParsers.js [hunk 8]
```diff
@@ -0,0 +1,451 @@
...
+                          pfConfigurationSyslogParserRegexRuleActions.reevaluate_access,
+                          pfConfigurationSyslogParserRegexRuleActions.update_ip4log,
+                          pfConfigurationSyslogParserRegexRuleActions.update_role_configuration,
+                          pfConfigurationSyslogParserRegexRuleActions.trigger_security_event,
+                          pfConfigurationSyslogParserRegexRuleActions.release_all_security_events,
+                          pfConfigurationSyslogParserRegexRuleActions.modify_node,
+                          pfConfigurationSyslogParserRegexRuleActions.close_security_event,
+                          pfConfigurationSyslogParserRegexRuleActions.register_node,
+                          pfConfigurationSyslogParserRegexRuleActions.dynamic_register_node
+                        ]
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogParsers.js:361]: Could we just use `Object.values(pfConfigurationSyslogParserRegexRuleActions)` instead?

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js [hunk 9]
```diff
@@ -0,0 +1,1019 @@
...
+  hasWrixLocations,
+  wrixLocationExists
+} from '@/globals/pfValidators'
+
+const {
+  maxLength,
+  required
+} = require('vuelidate/lib/validators')
+
+export const pfConfigurationWrixLocationsLogs = [
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:22]: Same thing as `pfConfigurationSyslogForwardersLogs`? Can we define this array globally? Will be replaced by the new *OPTIONS* API endpoints?

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js [hunk 10]
```diff
@@ -0,0 +1,1019 @@
...
+import pfFormRangeToggle from '@/components/pfFormRangeToggle'
+import {
+  pfConfigurationListColumns,
+  pfConfigurationListFields
+} from '@/globals/configuration/pfConfiguration'
+import {
+  and,
+  not,
+  conditional,
+  hasWrixLocations,
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js]: To make @extrafu happy, we'll need to replace *Wrix* by *WRIX* everywhere.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js [hunk 11]
```diff
@@ -0,0 +1,1019 @@
...
+                [i18n.t('Wrix Location exists.')]: not(and(required, conditional(isNew || isClone), hasWrixLocations, wrixLocationExists))
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('Provider Identifier'),
+          fields: [
+            {
+              key: 'Provider_Identifier',
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:143]: Why mixing snake case and title case? Please adjust all keys to snake case only.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js [hunk 12]
```diff
@@ -0,0 +1,1019 @@
...
+                      { value: 'Pacific/Port_Moresby', text: 'Port_Moresby' },
+                      { value: 'Pacific/Rarotonga', text: 'Rarotonga' },
+                      { value: 'Pacific/Tahiti', text: 'Tahiti' },
+                      { value: 'Pacific/Tarawa', text: 'Tarawa' },
+                      { value: 'Pacific/Tongatapu', text: 'Tongatapu' },
+                      { value: 'Pacific/Wake', text: 'Wake' },
+                      { value: 'Pacific/Wallis', text: 'Wallis' }
+                    ]
+                  }
+                ]
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:894]: I hope those will eventually be provided by the server .. ?

File: html/pfappserver/root/static.alt/src/views/Configuration/_components/IntegrationSection.vue [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+        <dl>
+            <dt v-t="'Firewall SSO:'"></dt> <dd>This allows PacketFence to let the firewall know who is using a particular IP address on the network. The firewall can then apply per-user or per-role policies.</dd>
+            <dt v-t="'Cisco Mobility Services Engine:'"></dt> <dd>This provides PacketFence geographical location information of endpoints on the network. You can then provide a different captive portal based on where the device is physically located, or position an endpoint on a specific map of a building.</dd>
+            <dt v-t="'Web Services:'"></dt> <dd>This allows PacketFence to expose some of its APIs through Web Services. See the PacketFence Developers Guide for API documentation.</dd>
+            <dt v-t="'Syslog Parsers:'"></dt> <dd>PacketFence can receive security event information from multiple solutions over syslog. You can configure which data input you want to accept from this module.</dd>
+            <dt v-t="'WRIX:'"></dt> <dd>This allows PacketFence to define Wireless Roaming Intermediary eXchange (WRIX) information on access points it manages.</dd>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Configuration/_components/IntegrationSection.vue]: Those strings are localized. Let's use them.

File: html/pfappserver/root/static.alt/src/views/Configuration/_components/WrixLocationView.vue [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+      </h4>
+    </template>
+    <template slot="footer"
+      scope="{isDeletable}"
+    >
+      <b-card-footer @mouseenter="$v.wrixLocation.$touch()">
+        <pf-button-save :disabled="invalidForm" :isLoading="isLoading">
+          <template v-if="isNew">{{ $t('Create') }}</template>
+          <template v-else-if="isClone">{{ $t('Clone') }}</template>
+          <template v-else-if="ctrlKey">{{ $t('Save &amp; Close') }}</template>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Configuration/_components/WrixLocationView.vue]: Is the HTML entity encoding really necessary?

**Review by cgx — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js (hunk 11 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:143]: Title case is used right through to the db schema (https://github.com/inverse-inc/packetfence/blob/devel/db/pf-schema-8.3.0.sql#L1001). Should we modify it there first (another PR)?

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js (hunk 9 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:22]: This is just a temporary placeholder. It's expected with OPTIONS endpoint(s).

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogParsers.js (hunk 8 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationSyslogParsers.js:361]: I prefer this - although verbose - as it allows alphabetical ordering in the definition, and static sorting in the reference. If we combine as you suggest we'll lose one of these benefits.

File: html/pfappserver/root/static.alt/src/views/Configuration/_components/WrixLocationView.vue (hunk 14 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/views/Configuration/_components/WrixLocationView.vue]: I'll test. This is just from &quot ;habit&quot ;.

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js (hunk 12 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:894]: I expect with OPTIONS

File: html/pfappserver/root/static.alt/src/views/Configuration/_store/sources.js [hunk 15]
```diff
@@ -33,6 +33,18 @@ const actions = {
       return response.items
     })
   },
+  getAuthenticationSourcesByType: ({ state, commit }, type) => {
+console.log('byType >', type)
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Configuration/_store/sources.js]: Remove

File: html/pfappserver/root/static.alt/src/views/Configuration/_store/sources.js [hunk 16]
```diff
@@ -33,6 +33,18 @@ const actions = {
...
+    const params = {
+      sort: 'id',
+      fields: ['id', 'description', 'class'].join(','),
+      type: type
+    }
+    return api.authenticationSources(params).then(response => {
+console.log('byType <', response)
```
Comment (satkunas) [inline html/pfappserver/root/static.alt/src/views/Configuration/_store/sources.js]: Remove

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js (hunk 11 above)
Comment (cgx) [reply html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationWrixLocations.js:143]: Does it matter? I though column names would be case insensitive.

**Review by cgx — APPROVED** [review]

**Pull request merged**